### ISSUE TITLE
Allow users to select which accounts to add during Ledger login

### ISSUE THREAD

**Issue by kcole16** [near/near-wallet#782]
### Overview
The current Ledger login implementation requires users to add all accounts associated with their Ledger at once. If they reject an account (confirming the "add key" TX), the login is cancelled. Users should be able to select which accounts they want to login with.

### Acceptance Criteria
- [ ] User can view all accounts associated with their Ledger
- [ ] User can select which accounts to login with
- [ ] User only needs to confirm the `addKey` TX for the accounts they've selected

**Comment by ilblackdragon** [comment]
Why does adding key required at all? Why not just query public keys and show all accounts with these keys as available?

**Comment by marcinbodnar** [comment]
@kcole16 @vgrichina 
I think the best way to implement this would be:
1. User confirms public key
2. User sees modal with:
 a) list of accounts with checkboxes to choose each account (additionally select all button)
 b) "enter accountId manually" button
3. If user chooses a) - modal with a list of chosen accounts appears and the user is able to confirm each account.
4. If user chooses b) - modal with accountId input appears, and he is able to enter accountId.

This might be a better solution, than the variant where the user sees all found accounts and can confirm or reject each on Ledger device because if he had 20 accounts, and want to add only a few of them he would have to click 20 times on the Ledger device anyway. With this change, it will be easier. Additionally, user can check if the account that he would like to add appears on the list, and try to add it manually if not.

What do you thing?

**Comment by vgrichina** [comment]
> Why does adding key required at all? Why not just query public keys and show all accounts with these keys as available?

This is needed to authorize access to metadata in contract helper. Which is necessary because nearcore lacks any place to store metadata about keys / accounts.

**Comment by vgrichina** [comment]
> b) "enter accountId manually" button

@marcinbodnar  this shouldn't be needed. We need to fix indexer @khorolets @frol 

> a) list of accounts with checkboxes to choose each account (additionally select all button)

I think we should select all accounts by default and allow user to uncheck specific ones if they don't want to recover all matching ones.

**Comment by marcinbodnar** [comment]
@vgrichina 
> this shouldn't be needed. We need to fix indexer

Sure, but it's ready, and can be used as an additional feature - if the user has 20 accounts, and want to add only one, instead of searching the list he can write it. It's nothing special, but since this functionality is ready it can be used as add-on.

**Comment by Patrick1904** [comment]
> @kcole16 @vgrichina
> I think the best way to implement this would be:
> 
> 1. User confirms public key
> 2. User sees modal with:
>    a) list of accounts with checkboxes to choose each account (additionally select all button)
>    b) "enter accountId manually" button
> 3. If user chooses a) - modal with a list of chosen accounts appears and the user is able to confirm each account.
> 4. If user chooses b) - modal with accountId input appears, and he is able to enter accountId.
> 
> This might be a better solution, than the variant where the user sees all found accounts and can confirm or reject each on Ledger device because if he had 20 accounts, and want to add only a few of them he would have to click 20 times on the Ledger device anyway. With this change, it will be easier. Additionally, user can check if the account that he would like to add appears on the list, and try to add it manually if not.
> 
> What do you thing?

@corwinharrell what do you think about proposed flow?

**Comment by ilblackdragon** [comment]
This is implemented now, isn't it? @stefanopepe

**Comment by marcinbodnar** [comment]
@ilblackdragon @stefanopepe It's implemented but in a bit different way than in the description or I proposed. The user is able to reject some accounts and accept only the one he would like to. Rejecting an account doesn't break the flow. But the user is not able to `select` the accounts he would like to add before going to the signing step - the flow is: reject/accept account one by one.

**Comment by corwinharrell** [comment]
yep just for some context @marcinbodnar this is a planned addition following the updates to Ledger. @MaximusHaximus has got to get the Ledger fixes shipped first and then this will be a follow up piece of work. I already have the UX figured out 👍
